cudax/include/cuda/experimental/__device/device.cuh

```diff
@@ -0,0 +1,117 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__DEVICE_DEVICE
+#define _CUDAX__DEVICE_DEVICE
+
+#include <cuda_runtime_api.h>
+// cuda_runtime_api needs to come first
+
+#include <cuda/std/__cuda/api_wrapper.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+namespace cuda::experimental
+{
+// Dummy struct for now to be able to reference it in other places
+// TODO this might be device_ref instead
+// TODO proper implementation
+//! @brief A non-owning representation of a CUDA device
+class device
+{
+  int __id = 0;
+
+public:
+  //! @brief Create a `device` object from a native device ordinal.
+  /*implicit*/ constexpr device(int id) noexcept
+      : __id(id)
+  {}
+
+  //! @brief Retrieve the native ordinal of the device
+  //!
+  //! @return int The native device ordinal held by the device object
+  _CCCL_NODISCARD constexpr int get() const noexcept
+  {
+    return __id;
+  }
+};
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
+
+//! @brief RAII helper which saves the current device and switches to the
+//!        specified device on construction and switches to the saved device on
+//!        destruction.
+//!
+struct __scoped_device
```

Comment:
This might seem redundant with a similar type in #2073, but I plan to do follow-up changes to have this type use driver API added in #2017, while the other one is in libcu++ and will continue to use runtime API for now